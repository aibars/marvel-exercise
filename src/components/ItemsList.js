import React from 'react';
import '../styles/ItemsList.css';

export default function ItemsList(props) {

    return (
        <div>
            <ul>
                {props.items.map((item, i) => (
                    <li key={i}>{item.name}</li>
                ))}
            </ul>
        </div>
    );
}