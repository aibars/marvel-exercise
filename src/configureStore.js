import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const loggerMiddleware = createLogger();

export default function configureStore(preloadedState) {
  const enhancers = [];
  const windowIfDefined = typeof window === 'undefined' ? null : window;
  if (windowIfDefined && windowIfDefined.__REDUX_DEVTOOLS_EXTENSION__) {
    enhancers.push(windowIfDefined.__REDUX_DEVTOOLS_EXTENSION__());
  }


  return createStore(
    rootReducer,
    preloadedState,
    compose(applyMiddleware(thunkMiddleware, loggerMiddleware), ...enhancers),
  );
}