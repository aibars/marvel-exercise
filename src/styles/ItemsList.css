.pokeball:hover {
    color: blue;
    cursor: pointer;
}

.items-list {
    font-size: calc(10px + 2vmin);
    margin-left: 20px;
    list-style-type: none;
}

.pokeball {
    background: url('./pokeball.ico') no-repeat left center;
    height: 40px;
    padding-left: 44px;
    padding-top: 3px;
}