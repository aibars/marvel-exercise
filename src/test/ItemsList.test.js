import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ItemsList from '../components/ItemsList';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('ItemsList', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            pokemons: {
                items: [{
                    name: '1',
                    url: 'url1'
                },
                {
                    name: '1',
                    url: 'url1'
                }],
                isFetching: false,
                lastUpdate: Date.now()
            }
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <ItemsList items={[{
                    name: '1',
                    url: 'url1'
                },
                {
                    name: '1',
                    url: 'url1'
                }]} />
            </Provider>
        );

    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should dispatch an action on clicking on a list element', () => {
        renderer.act(() => {
            component.root. findAllByType('li')[0].props.onClick();
        });
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
});