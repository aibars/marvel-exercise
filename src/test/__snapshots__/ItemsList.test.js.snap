// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList should render with given state from Redux store 1`] = `
<div>
  <ul
    className="items-list"
  >
    <li
      className="pokeball"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="pokeball"
      onClick={[Function]}
    >
      1
    </li>
  </ul>
</div>
`;
